trie/sync.go

```diff
@@ -458,7 +463,7 @@ func (s *Sync) commitNodeRequest(req *nodeRequest) error {
 	// Write the node content to the membatch
 	s.membatch.nodes[string(req.path)] = req.data
 	s.membatch.hashes[string(req.path)] = req.hash
-
```

Comment:
Could you doc this too? Very hard to follow and easy to miss if we ever need to update the way a node is stored

Reply:
Yeah I agree. I think in hash-based, we don't need to count the req.path since it's not a part of storage data. Maybe just `common.HashLength + len(req.data)`?

And we need to introduce the state scheme abstraction here. Like the PR https://github.com/ethereum/go-ethereum/pull/25532/files I can add one more API called `NodeSize` to return the approximate node size according to the scheme we use.